packages/amplify-provider-awscloudformation/src/iterative-deployment/deployment-state-manager.ts

```diff
@@ -97,14 +99,20 @@ export class DeploymentStateManager implements IDeploymentStateManager {
   public startCurrentStep = async (params?: StepStatusParameters): Promise<void> => {
     if (this.direction === 1) {
       if (this.getCurrentStep().status !== DeploymentStepStatus.WAITING_FOR_DEPLOYMENT) {
-        throw new Error(`Cannot start step then the current step is in ${this.getCurrentStep().status} status.`);
+        throw new AmplifyError('DeploymentError', {
+          message: `Cannot start step then the current step is in ${this.getCurrentStep().status} status.`,
```

Comment:
Looks like this typo appears several times in this file
```suggestion
          message: `Cannot start step when the current step is in ${this.getCurrentStep().status} status.`,
```